src/fenicsxconcrete/boundary_conditions/bcs.py

```diff
@@ -69,17 +71,17 @@ def __init__(
 
     def add_dirichlet_bc(
         self,
-        value: typing.Union[
-            dolfinx.fem.Function,
-            dolfinx.fem.Constant,
-            dolfinx.fem.DirichletBCMetaClass,
-            npt.NDArray,
-            typing.Callable,
-        ],
-        boundary: typing.Union[int, npt.NDArray, typing.Callable] = None,
+        value: (
+            dolfinx.fem.Function |
+            dolfinx.fem.Constant |
+            dolfinx.fem.DirichletBCMetaClass |
+            np.ndarray |
+            typing.Callable
+        ),
+        boundary: int | np.ndarray | typing.Callable | None = None,
```

Comment:
same as other comment regarding `np.ndarray` vs. `numpy.typing.NDArray` and if `| None` is necessary

Reply:
Also, I think we can just use `callable` without importing from `typing`